ios/RNMobileDeviceManager.m

```diff
@@ -0,0 +1,88 @@
+#import "RNMobileDeviceManager.h"
+
+// Used to send events to JS
+#if __has_include(<React/RCTBridge.h>)
+#import <React/RCTBridge.h>
+#elif __has_include("RCTBridge.h")
+#import "RCTBridge.h"
+#else
+#import "React/RCTBridge.h"
+#endif
+
+#if __has_include(<React/RCTEventDispatcher.h>)
+#import <React/RCTEventDispatcher.h>
+#elif __has_include("RCTEventDispatcher.h")
+#import "RCTEventDispatcher.h"
+#else
+#import "React/RCTEventDispatcher.h"
+#endif
+
+@implementation MobileDeviceManager
+
+@synthesize bridge = _bridge;
+
+static NSString * const appConfigurationKey = @"com.apple.configuration.managed";
+static NSString * const APP_CONFIG_CHANGED = @"react-native-mdm/managedAppConfigDidChange";
+
+- (instancetype)init
+{
+    if ((self = [super init])) {
+        // Add Notification Center observer to be alerted of any change to NSUserDefaults.
+        // Managed app configuration changes pushed down from an MDM server appear in NSUSerDefaults.
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(userDefaultsDidChange:) name:@"NSUserDefaultsDidChangeNotification" object:nil];
```

Comment:
What issues did you run into when observing specific keys in the `NSUserDefaults` observer?